src/contexts/events.tsx

```diff
@@ -143,12 +153,28 @@ const useEvents = () => {
     trackedEventTypes.current = new Set();
   }, []);
 
+  useEffect(() => {
+    if (!chainId) return;
+
+    if (previousChainId.current === undefined) {
+      previousChainId.current = chainId;
+      // This means we are in the first render, so we don't want to track the event
+      return;
+    }
+
+    trackEvent({
+      event: 'network_change',
+      from_network: previousChainId.current || chainId,
+      to_network: chainId,
+    });
+
+    previousChainId.current = chainId;
+  }, [chainId, previousChainId]);
```

Comment:
missing dependency: `trackEvent`

Reply:
`previousChainId` is a useRef and it doesn't change across renders. We don't have to include it in the dependency array